django_irods/storage.py

```diff
@@ -71,23 +72,41 @@ def zipup(self, in_name, out_name):
         :param out_name: the output zipped file name
         :return: None
         """
-        in_bucket, in_path = bucket_and_name(in_name)
+        def chunk_request(zip_archive_file, bucket, key):
+            chunk_size = 1024 * 1024 * 256  # 256MB
+            object_attrs = self.connection.meta.client.get_object_attributes(Bucket=bucket, Key=key,
+                                                                             ObjectAttributes=["ObjectSize"])
+            object_size = object_attrs.get("ObjectSize")
+
+            chunk_start = 0
+            chunk_end = chunk_start + chunk_size - 1
+
+            while chunk_start < object_size:
+                # Read specific byte range from file as a chunk. We do this because AWS server times out and sends
+                # empty chunks when streaming the entire file.
+                if body := self.connection.meta.client.get_object(
+                    Bucket=bucket, Key=key, Range=f"bytes={chunk_start}-{chunk_end}"
+                ).get("Body"):
+                    chunk = body.read()
+                    zip_archive_file.write(chunk)
+                    chunk_start += chunk_size
+                    chunk_end += chunk_size
+                    chunk_end = min(chunk_end, object_size)
+
+        in_bucket_name, in_path = bucket_and_name(in_name)
         out_bucket, out_path = bucket_and_name(out_name)
-        in_bucket = self.connection.Bucket(in_bucket)
+        in_bucket = self.connection.Bucket(in_bucket_name)
 
-        # Stream zip https://stackoverflow.com/a/69136133
         filesCollection = in_bucket.objects.filter(Prefix=in_path).all()
-        archive = BytesIO()
+
         in_prefix = os.path.dirname(in_path) if self.isDir(in_name) else in_path
-        with zipfile.ZipFile(archive, 'w', zipfile.ZIP_DEFLATED) as zip_archive:
-            for file in filesCollection:
-                relative_path = file.key[len(in_prefix):]
-                with zip_archive.open(relative_path, 'w', force_zip64=True) as file1:
-                    file1.write(file.get()['Body'].read())
-
-        archive.seek(0)
-        self.connection.Object(out_bucket, out_path).upload_fileobj(archive)
-        archive.close()
+        with open(f's3://{out_bucket}/{out_path}', 'wb', transport_params={'client': self.connection.meta.client}
+                  ) as out_file:
+            with zipfile.ZipFile(out_file, 'w', zipfile.ZIP_DEFLATED) as zip_archive:
```

Comment:
nice, I didn't know that you need to specify compression -- thought it was included by default!

Reply:
The default is ZIP_STORED